apps/gitness/src/pages-v2/repo/repo-summary.tsx

```diff
@@ -35,17 +36,27 @@ export default function RepoSummaryPage() {
   const { spaceId, repoId } = useParams<PathParams>()
   const [gitRef, setGitRef] = useState<string>('')
 
+  const {
+    branchList,
+    setBranchList,
+    setSpaceId,
+    setRepoId,
```

Comment:
Hmm. Why are these separate functions?
Since a branch would depend on space and repo, I think it should be one fn

Reply:
done